### PR TITLE
Introduce Platform (BOM) for version management

### PR DESCRIPTION
**Pull request by jjohannes**
Use a platform (aka BOM) for versions

This replaces the version catalog. A platform can be...
- published as BOM (understood by Gradle and Maven)
- used to control (e.g. upgrade) versions of transitive dependencies
- shared by multiple independent builds (hedera-node / platform-sdk)

This also:
- simplifies the project names as much as possible as another step to align project and module names more (see also #6918). For example:
  - `project(":hedera-node:node-app")` -> `project(":app")`
  - `project(":hedera-node:node-app-hapi-utils")` `project(":app-hapi-utils")`
- turns `buildSrc` into an independent so-called _included build_, which can then be used by multiple "builds" in the repository. The "platform" is a separate build that 'hedera-node' now includes and 'platform-sdk' will also include it with the next step.

**TODO: How to name the platform project/component?**
Right now it's `com.hedera.hashgraph:hedera-platform` and should probably be changed.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: settings.gradle.kts
Comment (jjohannes) [inline settings.gradle.kts:131]: Initially, I wanted to put these into the platform as well. But the 'protobuf' plugin is very stubborn in picking this up for the internal proto tool dependency resolution it does. I am sure I can find a way though if we absolutely want it.

However there are pro and cons, so I left it here for now to postpone this decision.

PRO (move to platform)
- everything in one place
- there probably is a relationship between protobuf modules (e.g. com.google.protobuf) and the protobuf compiler(s). So the version – 3.19.4 – is repeated.

CON (keep it here)
- The proto compilers are not Modules to they can't be addressed by module name. But it would be nice to use module names only in the platform. But it's not a must. (Or we patch the compilers only for the sake of having a name...)
- Mixing tool versions and library/module versions in one place. It's a bit of a philosophical question if this is "bad".

I think I would tend toward moving these into the platform as well eventually. But I would do that as a follow up then.

Comment (nathanklick) [reply settings.gradle.kts:131]: I agree with a follow up PR on this and with moving to the platform (BOM). We should also look at updating helidon to at least version 3.1.0 as well.

**Review by nathanklick — COMMENTED** [review]
Is moving `buildSrc` into an actual project as `build-logic` a short term solution or will this be a permanent change?  I am against it being a permanent change as we do not want this being published to maven central. 

Also, is it excluded from the maven central publish and will not publishing the `build-logic` module break the other modules in the interim?

**Comment by jjohannes** [comment]
Maybe another clarification:

Similar to how you declare repositories for "components" vs repositories for "gradle plugins", you can do "includeBuild" (things from source) for "components" vs "gradle plugins".

``` 
pluginManagement {
  repositories.gradlePluginPortal()
  includeBuild("build-logic") // thing providing Gradle plugins
}

includeBuild("hedera-platfrom") // another component for my product build from source

// I sometimes write the latter like this to make the destination clearer:
// dependencyResolutionManagement {
//.  repositories.mavenCentral()
//   includeBuild("hedera-platfrom")
// }

```

**Comment by nathanklick** [comment]
@jjohannes Does this bind us to publishing the `build-logic` module to Maven Central?

**Comment by jjohannes** [comment]
> Does this bind us to publishing the build-logic module to Maven Central?

**No**. As buildSrc, it's used during building. The published components are independent of it. Nothing changes in that regard.

**Comment by jjohannes** [comment]
The `hedera-platform` (the BOM) is what needs to be published (in the setup right now). That could also be done different depending on what we want.

I realize now that this PR is not complete in that regard. It does not yet configure everything to publish `hedera-platform` with all the information as the other components.

Following questions:
1. Is `hedera-platform` a good name?
2. Should it be published as BOM (would need to adjust `com.hedera.hashgraph.platform.gradle.kts` to contain parts of `com.hedera.hashgraph.maven-publish.gradle.kts`)
3. Instead of 2, we can configure the publishing of components to include the versions that were used ([publish resolved versions](https://docs.gradle.org/current/userguide/publishing_maven.html#publishing_maven:resolved_dependencies)) of their dependencies and **not publish a BOM** - which would from the publishing perspective be similar to how things are now (needs adjustment to `com.hedera.hashgraph.maven-publish.gradle.kts` for publishing resolved versions)

The publishing BOM topic can also be confusing. There are basically two types of BOMs.

1. A BOM that contains the versions of all the components the framework/library depends on (example  [spring-boot-dependencies](https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-dependencies/3.1.0/spring-boot-dependencies-3.1.0.pom))
2. A BOM that contains the versions of all my own components to help aligning them to one version. Usually it lists all the components that belong to a library/frameowk (exmaples: [JUnit-BOM](https://repo1.maven.org/maven2/org/junit/junit-bom/5.9.3/junit-bom-5.9.3.pom), [Jackson-BOM](https://repo1.maven.org/maven2/com/fasterxml/jackson/jackson-bom/2.15.2/jackson-bom-2.15.2.pom)) 

What we get with this PR would be a "BOM" of the first kind.

**Comment by jjohannes** [comment]
Looks like right now `hedera-evm` is the only component in `hedera-node` that has publishing configured. What's the overall state of the publishing? If it is not really used/stable, we could also merge the PR as it is and discuss more details in a follow up.

**Review by nathanklick — DISMISSED** [review]

**Comment by nathanklick** [comment]
> Looks like right now `hedera-evm` is the only component in `hedera-node` that has publishing configured. What's the overall state of the publishing? If it is not really used/stable, we could also merge the PR as it is and discuss more details in a follow up.

The Hedera EVM module is currently the only module published to Maven Central. Right now it uses an externally triggered pipeline, but I will be bringing the pipeline into the repository over the next several weeks.

**Comment by jjohannes** [comment]
@nathanklick thanks for clarifying. I updated this PR so that the publishing is kept as it is now:
-  Hedera EVM module is published with versions in the metadata as before
- The new hedera-platfrom is not published

Then, from my side, this PR can be merged and we can talk about publishing more in one of the next steps.

(two integration tests are failing right now, but they were good on this branch before developed was merged again – see https://github.com/hashgraph/hedera-services/pull/6965/commits/4bb72490f84bc8902bf99a191d288ad9fee71a5d – maybe this is a problem on develop already?)

**Review by nathanklick — DISMISSED** [review]

**Review by nathanklick — DISMISSED** [review]
Updated with the latest changes from the `develop` branch. I am good with merging if all the tests pass.

**Comment by jjohannes** [comment]
Looks good now. GH tells me "Code owner review required", so I can't hit the merge button myself right now.

File: hedera-node/hedera-network-admin-service-impl/build.gradle.kts
Comment (netopyr) [inline hedera-node/hedera-network-admin-service-impl/build.gradle.kts:27]: This is wrong. The NetworkAdmin-Service should not require access to Token-Service implementation. If this wasn't just added by accident, we need to investigate and clean this up.

**Review by netopyr — APPROVED** [review]
LGTM in regards to the hedera-services files.

File: hedera-node/hedera-network-admin-service-impl/build.gradle.kts
Comment (jjohannes) [reply hedera-node/hedera-network-admin-service-impl/build.gradle.kts:27]: This was reported by the dependency analysis we have now – `./gradlew checkAllModuleInfo`. To make sure this is still there, I ran the analysis again and addressed all additional issues it found.

This one is still there. You can do 
`./gradlew :app-service-network-admin-impl:reason --id :app-service-token-impl`
to get information why:

```
Source: test
------------
* Uses 3 classes: com.hedera.node.app.service.token.impl.ReadableAccountStoreImpl, com.hedera.node.app.service.token.impl.ReadableTokenRelationStoreImpl, com.hedera.node.app.service.token.impl.ReadableTokenStoreImpl (implies testImplementation).
```

That should probably be change then.

**Comment by jjohannes** [comment]
Thanks @netopyr. I integrated latest develop and did the dependency analysis again (see other comment). I think issues with the dependencies should be done in follow-ups. Maybe you can take a note of the test dependency issue in `hedera-network-admin-service-impl` or plan in that someone knowing that part of the code base takes a look.

Could you give your approval for this PR again?

File: hedera-node/hedera-network-admin-service-impl/build.gradle.kts
Comment (netopyr) [reply hedera-node/hedera-network-admin-service-impl/build.gradle.kts:27]: Thanks, I will create an issue.

**Review by netopyr — APPROVED** [review]
LGTM in regards to hedera-services

**Pull request merged**
